### PR TITLE
Add reference check by alias and team [SUBS-501]

### PR DESCRIPTION
**Pull request by fpenim**
**WARN**
This is currently using the last working version of the `validator-common`:
[validator-common:48c266b9b0](https://github.com/EMBL-EBI-SUBS/validator-common/tree/48c266b9b0216cdba7f092cc91038760595a0e3a)

Updates:
- Added tests for all reference validators
- Added missing messaging configurations
- Added reference check by alias and team

### CODE REVIEW

**Comment by fpenim** [comment]
**WARN**
This is now using the `validator-common-1.2.0-SNAPSHOT`

Please ignore failing codacy check.

File: src/main/java/uk/ac/ebi/subs/validator/core/validators/SampleRefValidator.java [hunk 1]
```diff
@@ -15,7 +15,10 @@
 
 @Service
 @EnableMongoRepositories(basePackageClasses = SampleRepository.class)
-public class SampleRefValidator implements ReferenceValidator {
+public class SampleRefValidator extends AbstractReferenceValidator {
+
+    String FAIL_TEAM_AND_ALIAS_MESSAGE = "Could not find reference for ALIAS: %s in TEAM: %s";
+    String FAIL_MESSAGE = "Could not find reference target: %s .";
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/core/validators/SampleRefValidator.java]: These constants are already defined in the AbstractReferenceValidator

File: src/test/java/uk/ac/ebi/subs/validator/core/validators/AssayRefValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        when(assayRepository.findFirstByTeamNameAndAliasOrderByCreatedDateDesc("Team", "Alias")).thenReturn(new Assay());
+    }
+
+    @Test
+    public void assayFoundTest() {
+        AssayRef assayRef = generateAssayRef("ASSAY1234");
+        SingleValidationResult singleValidationResult = generateSingleValidationResult();
+
+        assayRefValidator.validate(assayRef, singleValidationResult);
+        System.out.println(singleValidationResult.getMessage());
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/validator/core/validators/AssayRefValidatorTest.java:39]: Do we need the `System.out.println()` lines?

**Review by ke4 — CHANGES_REQUESTED** [review]
Only small things :)

File: src/test/java/uk/ac/ebi/subs/validator/core/validators/AssayRefValidatorTest.java (hunk 2 above)
Comment (fpenim) [reply src/test/java/uk/ac/ebi/subs/validator/core/validators/AssayRefValidatorTest.java:39]: No we don't **need** it. But it's a visual help for the test results.

**Review by ke4 — APPROVED** [review]

**Pull request merged**